spring-data-neo4j-rx/src/main/java/org/neo4j/springframework/data/core/mapping/DefaultNeo4jPersistentEntity.java

```diff
@@ -276,4 +298,42 @@ private IdDescription computeIdDescription() {
 		return Collections.unmodifiableCollection(computedGraphProperties);
 	}
 
+	@Override
+	public Collection<GraphPropertyDescription> getGraphPropertiesInHierarchy() {
+
+		TreeSet<GraphPropertyDescription> allPropertiesInHierarchy =
+			new TreeSet<>(Comparator.comparing(GraphPropertyDescription::getPropertyName));
+
+		allPropertiesInHierarchy.addAll(getGraphProperties());
+		for (NodeDescription<?> childNodeDescription : getChildNodeDescriptionsInHierarchy()) {
+			Collection<GraphPropertyDescription> childGraphProperties = childNodeDescription.getGraphProperties();
+			allPropertiesInHierarchy.addAll(childGraphProperties);
+		}
+
+		return Collections.unmodifiableCollection(allPropertiesInHierarchy);
```

Comment:
```suggestion
		return allPropertiesInHierarchy;
```

It's a copy anyway… Not local state.